semgrep_metrics.atd

```diff
@@ -129,9 +133,16 @@ type extension = {
 (* Misc *)
 (*****************************************************************************)
 
+type pro_features = {
+    (* The value is determined by the `--diff-depth` option, and it's utilized
+       for inter-file differential scanning. Since semgrep 1.46 *)
+    ?diffDepth <ocaml mutable>: int option;
+}
+
 type misc = {
     (* coupling: features is commented a lot in semgrep/PRIVACY.md *)
     features <ocaml mutable>: string list;
+    proFeatures <ocaml mutable>: pro_features option;
```

Comment:
You need the '?' prefix, otherwise if the backend uses this interface file to process the metrics, it will crash
when encountering metrics generated by old version of semgrep.
See https://atd.readthedocs.io/en/latest/atdgen-tutorial.html#smooth-protocol-upgrades

Reply:
So is it more preferable to use `?proFeatures : pro_features option` rather than `~proFeatures : pro_features`? The linked document seems to use `~z : int` for adding a new optional field.

Reply:
I think @mjambon prefers ~, and it seems backward compatible too. I tend to use ? by habits.

Reply:
Updated.